### PR TITLE
[RFR] Re-add eslint recommended rules and fix errors

### PR DESCRIPTION
**Pull request by fzaninotto**
Follows #821

### CODE REVIEW

File: src/mui/form/FormTab.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React from 'react';
 import FormField from './FormField';
 
-const FormTab = ({ label, icon, children, ...rest }) =>
+const FormTab = (
+    { label, icon, children, ...rest } // eslint-disable-line no-unused-vars
```
Comment (djhi) [inline src/mui/form/FormTab.js]: Can be configured like this to allow using rest siblings as a way to filter items:

```js
"no-unused-vars": ["error", { "ignoreRestSiblings": true }]
```

Comment (fzaninotto) [inline src/mui/form/FormTab.js]: Didn't know that, thanks!

**Comment by fzaninotto** [comment]
Code review applied

**Comment by fzaninotto** [comment]
Wait, I have one more rule to add

**Comment by fzaninotto** [comment]
Now we're good

**Pull request merged**